common/scala/src/main/scala/whisk/core/containerpool/Container.scala

```diff
@@ -189,6 +192,9 @@ trait Container {
         RunResult(Interval(started, finished), response)
       }
   }
+  private def closeConnections(): Future[Unit] = {
+    Future.successful(httpConnection.foreach(_.close()))
+  }
```

Comment:
Should we break the `AutoClosable` usage (I know, I'm guilty here) and change the interface of `ContainerClient`'s `close` method to be `Future[Unit]`? That way, we can make use of akka's asynchronous closing without blocking the thread on that operation. Apache would of course still be blocking. WDYT?

Edit: Reading through some of the code, this is actually a `materializer.close()` for the Akka based client. That means we'll churn through Materializers in certain scenarios. Can you run the test suites against this change to see if that has any impact? IIRC Materializers are rather expensive to create.

Reply:
Looked through it briefly:

For the Apache based client, the ConnectionManager has a `closeIdleConnections()` method, which's timeout you can set to 0, so it should force-close all connections.

For the Akka based client, you need to get access to the `Http.HostConnectionPool`. You'll then have access to its underlying gateway which has a `shutdown` and a corresponding `startPool` method, which is documented to play just nice in tandem with `shutdown`. 
Alternatively, we can opt to just regenerate the hostConnectionPool itself for this case, if we think that's the safer alternative.

Measuring will tell if this is necessary optimisation though.